### PR TITLE
Add Support for AWS X-Ray Tracing Propagator

### PR DESCRIPTION
**Pull request by scottmace**
Adds support for the OpenTelemetry AWS X-Ray tracing [propagator](https://docs.rs/opentelemetry-aws/latest/opentelemetry_aws/).

This propagator helps propagate tracing information from upstream services (such as AWS load balancers) to downstream services and handles conversion between the X-Ray trace id format and OpenTelemetry span contexts.

**Checklist**

Complete the checklist (and note appropriate exceptions) before a final PR is raised.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]. It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]. Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]. Tick whichever testing boxes are applicable. If you are adding Manual Tests:
    - please document the manual testing (extensively) in the Exceptions.
    - please raise a separate issue to automate the test and label it (or ask for it to be labeled) as `manual test`

### CODE REVIEW

File: docs/source/configuration/tracing.mdx [hunk 1]
```diff
@@ -81,6 +81,9 @@ telemetry:
       # https://zipkin.io/ (compliant with opentracing)
       zipkin: false
 
+      # https://aws.amazon.com/xray/ (compliant with opentracing)
```
Comment (abernix) [inline docs/source/configuration/tracing.mdx:84]: > (compliant with opentracing)

I'm actually curious what this comment means (even on the other options).  Is it _compatible_?  What makes it so?

**Review by abernix — COMMENTED** [review]
Thanks for opening this.  I actually came across another issue about this today and finally got around to leaving my comment on it.  Maybe you could respond there? https://github.com/apollographql/router/issues/3445#issuecomment-1679429502

Other than that, one comment within here, though as my question on the other issue will elude to, I'm curious if this could just be done with the existing "custom header" option that exists and how this differs, in practice.

Thanks again!

File: docs/source/configuration/tracing.mdx (hunk 1 above)
Comment (scottmace) [reply docs/source/configuration/tracing.mdx:84]: I _think_ that It means that the format is compatible with the OpenTelemetry OTLP exporter/collector.

Comment (abernix) [reply docs/source/configuration/tracing.mdx:84]: "OpenTelemetry" makes sense, but I was more curious about "OpenTracing" (which is different, as I understand it)

**Comment by abernix** [comment]
Note that I think the tests are failing because of the need to commit the changes after reviewing snapshot changes with `cargo insta test`.  (You'd have to `cargo install cargo-insta` first, to use `cargo insta test`)

**Review by bnjjj — APPROVED** [review]

**Review by o0Ignition0o — APPROVED** [review]

**Comment by bnjjj** [comment]
@scottmace Thanks ! It looks good! Sorry for the delay, could you generate a small changeset for that pr ? Using `cargo xtask changeset create` please

**Pull request merged**